### PR TITLE
Adding rules

### PR DESCRIPTION
**Pull request by Chiamakabrowneyes**
## Description

Adding the rules component: 
Demo --> https://www.loom.com/share/b0b247a4fcca4027826011bd340af077 

## Link to Ticket: 

Ticket: https://www.notion.so/bathientran/Rules-Page-181977aa180e4cdcbcfb04c5cf314117?pvs=4

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## What Has Changed

Please describe in bullets what you have changed.

- Add rule section: 
- Added new route app/rules
- Allows users to edit and create new rules

## How Has This Been Tested / How Can This Be Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

Run the dev page and ensure that theres no bugs. Should be able to see the UI for the rules in "app/rules" route.
image

Made it adaptable to mobile:
image

### CODE REVIEW

**Comment by vickipetrova** [comment]
@Chiamakabrowneyes you should fix the deployment -- there's a check that fails, if we merge it rn it will break the deployment.

**Comment by vickipetrova** [comment]
> @Chiamakabrowneyes you should fix the deployment -- there's a check that fails, if we merge it rn it will break the deployment.

Actually, the break wasn't from this PR. It was from a failed migration directly applied to the production database. I fixed that issue so now all checks go through. Let's leave this PR open for now. As we discussed in the retrospective today, the rules feature is not part of the MVP and we are prioritizing the MVP tasks for the next two weeks. If we have time and users request this type of feature we can extend it so that -- there's a way to see the rules, add them, edit them, delete them, etc.

**Comment by Chiamakabrowneyes** [comment]
@vickipetrova @bathientran. 
Since we are done with the main MVP features. I've worked on the the backend functionailty for this rules component. 
image
image

File: src/app/_components/Login.js
Comment (bathientran) [inline src/app/_components/Login.js:1]: Have you tried testing this change? We don't actually use this component anymore.

**Review by bathientran — CHANGES_REQUESTED** [review]

File: src/app/_components/Login.js
Comment (Chiamakabrowneyes) [reply src/app/_components/Login.js:1]: Hey yeah, I tested it by running the app locally. It seems to work fine. 
I was reversing the changes that I had made, and that one seemed to pop up so I took it back to what I thought was its original version. 
The rules component doesn't directly use it though.

Comment (bathientran) [reply src/app/_components/Login.js:1]: If it isn't part of the rules feature, do you mind removing it then? makes our PRs cleaner and doesn't add tech debt to our repo

Comment (Chiamakabrowneyes) [reply src/app/_components/Login.js:1]: ALright on it

Comment (Chiamakabrowneyes) [reply src/app/_components/Login.js:1]: Okay done, Can you add back the migrations. Cause I had deleted it locally (and I saw that you added in the initial commit)

**Comment by bathientran** [comment]
Also you shouldn't be deleting the migrations, this would break our app, I fixed it in a previous commit. Could you add those back as well please? Thanks

**Comment by bathientran** [comment]
Same for the local .env file don't delete that file

**Comment by Chiamakabrowneyes** [comment]
I didnt delete the env file. Both of them are still present locally. idk why it shows as deleted. 

But yeah, I've added back the migrations. I had initially deleted it while trying to resolve the database error from earl;ier today

**Review by bathientran — APPROVED** [review]

**Pull request merged**